rust/src/delta.rs

```diff
@@ -839,14 +869,43 @@ impl DeltaTable {
 
     /// Returns provenance information, including the operation, user, and so on, for each write to a table.
     /// The table history retention is based on the `logRetentionDuration` property of the Delta Table, 30 days by default.
-    pub fn history(
+    /// If `limit` is given, this returns the information of the latest `limit` commits made to this table. Otherwise,
+    /// it returns all commits from the earliest commit.
+    pub async fn history(
         &mut self,
         limit: Option<usize>,
     ) -> Result<Vec<Map<String, Value>>, DeltaTableError> {
-        let commit_infos_list = self.state.commit_infos().iter().rev().map(Map::clone);
-        match limit {
-            Some(l) => Ok(commit_infos_list.take(l).collect()),
-            None => Ok(commit_infos_list.collect()),
+        let mut version = match limit {
+            Some(l) => max(self.get_latest_version().await? - l as i64 + 1, 0),
```

Comment:
nit, I think using `self.version` here should be good enough so the returned history is consistent with the current table state. If the user wants the latest history, he/she can manually call the update method to catch up.

Reply:
Fixed. Thanks.